web/utils.py

```
@@ -414,6 +416,288 @@ def create_leaderboard_context(
+
+            return customer
+
+        # Get existing customer
+        return stripe.Customer.retrieve(user.membership.stripe_customer_id)
+
+    except stripe.error.StripeError:
+        logger.exception(f"Error retrieving Stripe customer for user {user.email}")
+        return None
+
+
+def create_subscription(user: "User", plan_id: int, payment_method_id: str, billing_period: str) -> Dict[str, Any]:
```

Ruff (0.8.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)